### PR TITLE
Improve LDAP documentation and include a practical example with Futurama LDAP demo auth server 

### LINKED ISSUE #2938: Improve LDAP authentication support
Scouts support for LDAP authentication is fairly basic and the login documentation contains some minor errors.

It would be nice to describe how to use the authentication variables, LDAP_BIND_USER_DN  and LDAP_BIND_USER_PASSWORD and filtering variables LDAP_USER_DN and LDAP_GROUP_DN.

Also it would be good if we could mirror the function of `LDAP_REQUIRED_GROUP` so you could setup different user groups which has access to different softwares.

**Errors**
- Configuration variable `LDAP_REQUIRED_GROUP` is not supported by [flask-ldap3-login](https://flask-ldap3-login.readthedocs.io/en/0.9.16/configuration.html) flask-ldap3-login
- The documentation say that Scout supports login with username and password which is wrong. Scouts basic auth uses only email.

### LINKED ISSUE #3056: Change library used for LDAP auth to use one that allows manual auth steps
Regain control over the login to display any type of error messages.

### PR DESCRIPTION
**Pull request by northwestwitch**
- Use lib flask-ldapconn instead of flask_ldap3_login> to handle ldap authentication - to better handle connection errors (fix #3056)
- Improve documentation and fixing minor things in the login setup documentation (fix #2938)
- Step-to-step description on how to set up the LDAP login system in Scout using the demo LDAP server [docker-test-openldap](https://github.com/rroemhild/docker-test-openldap)

**How to test**:
1. Start a dockerized LDAP demo server containing the characters of futurama:
```
docker pull rroemhild/test-openldap
docker run --rm -p 10389:10389 -p 10636:10636 rroemhild/test-openldap
```
2. Choose a futurama character from those available on the server. Note that each character has an email and a password, Those will be used for the login in Scout. For instance let's take [Leela](https://github.com/rroemhild/docker-test-openldap#cnturanga-leelaoupeopledcplanetexpressdccom)
Leela
3. Let's save a User for Leela in Scout:
`scout --demo load user -i cust000 -u "Leela Turanga" -m leela@planetexpress.com -id leela@planetexpress.com`
4. Modify the demo config in scout to be able to connect to the LDAP server, by adding the following lines:
```
LDAP_HOST = "localhost"
LDAP_PORT = 10389
LDAP_BASE_DN = "dc=planetexpress,dc=com"
LDAP_USER_LOGIN_ATTR = "mail"
LDAP_USE_SSL = False
LDAP_USE_TLS = True
```
5. Start the demo scout server and log in as Leela
6. Bonus test: try to connect as another Futurama character that is not already present in the Scout database.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
Ready for review!

**Comment by northwestwitch** [comment]
Hi @mikaell, I noticed that you had very little time to work on Scout (which is totally fine, I know you are busy with your studies) so I was wondering if it's fine that this PR is reviewed by someone else? You could always reassign to it if you'll have time in the future?

**Comment by northwestwitch** [comment]
This is now fixed and ready for review. The main change I've introduced is the use of another library for authenticating to LDAP. This makes the process more flexible and allows a better control over the process so it's more easy to identify problems whenever the login is not working.

File: scout/server/blueprints/login/controllers.py [hunk 1]
```diff
@@ -1,20 +1,35 @@
-"""Code for login controllers"""
-
-DETECTIVES = [
```
Comment (dnil) [inline scout/server/blueprints/login/controllers.py:3]: Aw, now we never will know what M intended this for. 😸

File: scout/server/config.py [hunk 2]
```diff
@@ -21,6 +21,15 @@
 # If null no badge is displayed in scout
 ACCREDITATION_BADGE = "swedac-1926-iso17025.png"
 
+# LDAP login Settings
```
Comment (dnil) [inline scout/server/config.py:24]: 👍

File: scout/server/extensions/ldap_extension.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+from ldap3 import ALL, SYNC, Connection, Server, Tls
+
+LOG = logging.getLogger(__name__)
+
+
+class LdapManager(LDAPConn):
+    """Interface to LDAP login"""
+
+    def init_app(self, app):
+        ssl_defaults = ssl.get_default_verify_paths()
```
Comment (dnil) [inline scout/server/extensions/ldap_extension.py:14]: 👍

File: scout/server/blueprints/public/templates/public/index.html [hunk 4]
```diff
@@ -24,18 +24,17 @@ <h1 class="display-4">Scout</h1>
...
-            <label>Username{{ form.username(class="form-control") }}</label>
+      {% elif config.LDAP_HOST or config.LDAP_SERVER %}
+        <form class="form-row" method="POST" action="{{ url_for('login.login') }}" class="form-inline">
+          <div class="col-5">
+          <input type="text" name="ldap_user" class="form-control" placeholder="user email/id" required>
           </div>
-          <div class="col-4">
-            <label>Password{{ form.password(class="form-control") }}</label>
+          <div class="col-5">
+            <input type="text" name="ldap_password" class="form-control" placeholder="user password" required>
```
Comment (dnil) [inline scout/server/blueprints/public/templates/public/index.html:33]: Heh, we probably would need yet another for the hospital AD system which is more like a 2FA card with certificates system, but this should be similar enough to the old solution at least, while being more testable.

File: scout/server/extensions/ldap_extension.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            else ssl.CERT_NONE,
+            version=app.config["LDAP_TLS_VERSION"],
+            ca_certs_file=app.config["LDAP_CA_CERTS_FILE"],
+            valid_names=app.config["LDAP_VALID_NAMES"],
+            ca_certs_path=app.config["LDAP_CA_CERTS_PATH"],
+            ca_certs_data=app.config["LDAP_CA_CERTS_DATA"],
+            local_private_key_password=app.config["LDAP_PRIVATE_KEY_PASSWORD"],
+        )
+
+        self.ldap_server = Server(
```
Comment (dnil) [inline scout/server/extensions/ldap_extension.py:57]: The client represents the server.. Oh well, I guess that is fine - and not your doing! 😁

**Review by dnil — APPROVED** [review]

**Comment by dnil** [comment]
Works just fine now! Leela ok after adding: 
Screenshot 2021-12-20 at 11 29 33
but not the unauthorized professor:
Screenshot 2021-12-20 at 11 31 20

**Pull request merged**